zerver/tests/test_auth_backends.py

```diff
@@ -1652,8 +1719,48 @@ def register_extra_endpoints(self, requests_mock: responses.RequestsMock,
 
         self.email_data = email_data
 
-    def get_account_data_dict(self, email: str, name: str) -> Dict[str, Any]:
-        return dict(email=email, name=name)
+    def get_account_data_dict(self, email: str, name: str, user_avatar_url: str=None) -> Dict[str, Any]:
+        return dict(email=email, name=name, avatar_url=user_avatar_url)
+
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_register_new_invited_user_and_use_github_avatar(self) -> None:
+        """
+        This tests the registration flow in the case where an invitation for the user
+        was generated.
+        """
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        realm = get_realm('zulip')
+        mock_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+
+        iago = self.example_user('iago')
+        do_invite_users(iago, [email], [])
+
+        account_data_dict = self.get_account_data_dict(email=email, name=name, user_avatar_url=mock_user_avatar_url)
+        result = self.social_auth_test(account_data_dict,
+                                       expect_choose_email_screen=True,
+                                       subdomain=subdomain, is_signup='1')
+        self.stage_two_of_registration(result, realm, subdomain, email, name, name,
+                                       self.BACKEND_CLASS.full_name_validated,
+                                       user_avatar_url=mock_user_avatar_url, use_social_avatar='on')
+
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_register_new_user_and_use_github_avatar(self) -> None:
+        """Test adding support for a new user to register and use their github avatar.
+        If the user doesn't exist yet, social auth can be used to register an account"""
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        mock_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+        realm = get_realm("zulip")
+        account_data_dict = self.get_account_data_dict(email=email, name=name, user_avatar_url=mock_user_avatar_url)
+        result = self.social_auth_test(account_data_dict,
+                                       expect_choose_email_screen=True,
+                                       subdomain=subdomain, is_signup='1')
+        self.stage_two_of_registration(result, realm, subdomain, email, name, name,
+                                       self.BACKEND_CLASS.full_name_validated, user_avatar_url=mock_user_avatar_url,
+                                       use_social_avatar='on')
```

Comment:
These tests look good, but I think I don't agree with the implementation of details inside the helper functions :thinking: I'll post more comments on those. The only thing I'd change here is that `use_social_avatar` should probably be `True/False` and it should be an implementation detail of the helper function that `'on'` is the valued POSTed to the registration endpoint.

Reply:
Yeah, it makes more sense. I always kept mistyping this and writing `True` instead of `'on'`.